### DISCUSSION TITLE
Multiple views dynamically change size

### DISCUSSION THREAD

**Discussion by radoslavirha** [visgl/deck.gl#7156] [Q&A] [answered]
Hello,

Recently I was playing with multiple views. When I create fixed minimap, everything works fine. But I experience some weird behavior when I dynamically resize those views. After dynamic resize, when I zoom, it does not follow my cursor (zooming anchor point is somewhere else). Don't know how you call that point :sweat_smile:

Simply I have som splitter and it triggers method where I set new set of views:

```
const views = [
  new MapView(this.mainMapViewProps({
    x: '0%',
    y: '0%',
    height: '100%',
    width: `${value}%`
  })),
  new MapView(this.miniMapViewProps({
    x: `${value}%`,
    y: '0%',
    height: '100%',
    width: `${100 - value}%`
  }))
];

this.deckInstance.setProps({ views });
```

`this.mainMapViewProps` and `this.miniMapViewProps`  just set id + controller

I'm synchronising views in `onViewStateChange` but it seems there is no problem...it works fine with fixed views.

I have no idea what is wrong. Do you have any ideas? I can try to create simple example without my splitter just with + - buttons.

Thanks

https://user-images.githubusercontent.com/23722745/179994169-177f4b4e-91d6-4580-9433-cae34207f193.mov

**Comment by radoslavirha** [comment, marked as answer]
I created simple codesandbox: https://codesandbox.io/s/deck-gl-tilelayer-osm-forked-zkecqm?file=/src/App.js

**Reply by Pessimistress** [reply]
Opened https://github.com/visgl/deck.gl/pull/7208

**Reply by radoslavirha** [reply]
@Pessimistress works fine! Thank you!

**Reply by neelduttahere** [reply]
Hey @radoslavirha Can you repost the sandbox? Currently, it is not available at codesandbox.

**Comment by alienatorZ** [comment]
Do you know how to split the map similar to: [http://www.itowns-project.org/itowns/examples/#effects_split]?

**Reply by kylebarron** [reply]
You can have multiple views, one on each half of the webpage https://deck.gl/docs/developer-guide/views
